src/NServiceBus.Core.Tests/Routing/RoutingSettingsTests.cs

```diff
@@ -0,0 +1,148 @@
+            var result = await routingTable.GetDestinationsFor(new List<Type>
+            {
+                typeof(SomeMessageType)
+            }, new ContextBag());
+
+            Assert.That(result.Count(), Is.EqualTo(1));
+            var routingTargets = await RetrieveRoutingTargets(result);
+            Assert.That(routingTargets.Single().Endpoint, Is.EqualTo("destination"));
+        }
+
+        [Test]
+        public async Task WhenRoutingAssemblyToEndpoint_ShouldConfigureAllContainedMessagesInRoutingTable()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+
+            routingSettings.RouteToEndpoint(Assembly.GetExecutingAssembly(), "destination");
+
+            var routingTable = routingSettings.Settings.Get<UnicastRoutingTable>();
+            var result = await routingTable.GetDestinationsFor(new List<Type>
+            {
+                typeof(SomeMessageType),
+                typeof(OtherMessageType),
+                typeof(MessageWithoutNamespace)
+            }, new ContextBag());
+
+            Assert.That(result.Count(), Is.EqualTo(3));
+            var routingTargets = await RetrieveRoutingTargets(result);
+            Assert.That(routingTargets, Has.All.Property("Endpoint").EqualTo("destination"));
+        }
+
+        [Test]
+        public async Task WhenRoutingAssemblyWithNamespaceToEndpoint_ShouldOnlyConfigureMessagesWithinThatNamespace()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+
+            routingSettings.RouteToEndpoint(Assembly.GetExecutingAssembly(), nameof(MessageNamespaceA), "destination");
+
+            var routingTable = routingSettings.Settings.Get<UnicastRoutingTable>();
+            var result1 = await routingTable.GetDestinationsFor(new List<Type>
+            {
+                typeof(SomeMessageType)
+            }, new ContextBag());
+            Assert.That(result1.Count(), Is.EqualTo(1));
+
+            var result2 = await routingTable.GetDestinationsFor(new List<Type>
+            {
+                typeof(OtherMessageType),
+                typeof(MessageWithoutNamespace)
+            }, new ContextBag());
+            Assert.That(result2.Count(), Is.EqualTo(0));
+        }
+
+        [Test]
+        public async Task WhenRoutingAssemblyWithNamespaceToEndpointAndSpecifyingNullNamespace_ShouldOnlyConfigureMessagesWithinEmptyNamespace()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+
+            routingSettings.RouteToEndpoint(Assembly.GetExecutingAssembly(), null, "destination");
+
+            var routingTable = routingSettings.Settings.Get<UnicastRoutingTable>();
+            var result1 = await routingTable.GetDestinationsFor(new List<Type>
+            {
+                typeof(MessageWithoutNamespace)
+            }, new ContextBag());
+            Assert.That(result1.Count(), Is.EqualTo(1));
```

Comment:
I totally understand that we are using `Count()`. But still there is a caveat. We would not catch if the wrong destination would be returned. So should we assert the destinations instead?